rotkehlchen/db/eth2.py

```diff
@@ -163,20 +165,44 @@ def get_validators(self, cursor: 'DBCursor') -> list[ValidatorDetails]:
         )
         return [ValidatorDetails.deserialize_from_db(x) for x in cursor]
 
+    def get_consolidated_validators(self, cursor: 'DBCursor') -> dict[int, int]:
+        """Returns a mapping of source validator index to target validator index
+        for all validators that have been consolidated.
+        """
+        cursor.execute(
+            'SELECT e.extra_data FROM history_events AS e LEFT JOIN evm_events_info ON evm_events_info.identifier=e.identifier '  # noqa: E501
+            'WHERE evm_events_info.counterparty = ? AND e.type = ? AND e.subtype = ? ',
+            (CPT_ETH2, HistoryEventType.INFORMATIONAL.serialize(), HistoryEventSubType.CONSOLIDATE.serialize()),  # noqa: E501
+        )
+        consolidation_indices = {}
+        for raw_extra_data in cursor:
+            try:
+                extra_data = json.loads(raw_extra_data[0])
+                consolidation_indices[extra_data['source_validator_index']] = extra_data['target_validator_index']  # noqa: E501
+            except (KeyError, json.JSONDecodeError) as e:  # should never happen
+                log.error(f'Unable to decode extra data from eth2 consolidation event {raw_extra_data} due to {e}')  # noqa: E501
+                continue
+
+        return consolidation_indices
+
     def get_validators_with_status(
             self,
             cursor: 'DBCursor',
             validator_indices: set[int] | None,
     ) -> list[ValidatorDetailsWithStatus]:
         result: list[ValidatorDetailsWithStatus] = []
         exited_indices = self.get_exited_validator_indices(cursor)
+        consolidated_indices = self.get_consolidated_validators(cursor)
```

Comment:
question: notice how `validator_indices` is not used in the query and is filtered later at the python level. Why not use them in the queries if the set is not None?

Reply:
> question: notice how validator_indices is not used in the query and is filtered later at the python level. Why not use them in the queries if the set is not None?

the validator indices are in the `extra_data` as json string so filtering won't be straight-forward.